Sources/qiskit/datastructures/Matrix.swift

```diff
@@ -443,6 +457,74 @@ extension Matrix where T == Complex {
         return m
     }
 
+    public func eigh() throws -> (Vector<Double>, Matrix<Complex>) {
```

Comment:
As mentioned in https://github.com/QISKit/qiskit-sdk-swift/pull/5, although this method is based on LAPACK function `zheevd`, it is not as efficient as the latter because it iterates up to 6 times through the entire list of values and its memory footprint is bigger.